pkg/service/init.go

```diff
@@ -1,57 +1,115 @@
 package service
 
 import (
-	"fmt"
+	"context"
 	"io"
 	"os"
-	"os/signal"
 	"runtime/debug"
-	"syscall"
+	"sync"
 
 	"github.com/sirupsen/logrus"
 
 	ausf_context "github.com/free5gc/ausf/internal/context"
 	"github.com/free5gc/ausf/internal/logger"
+	"github.com/free5gc/ausf/internal/sbi"
 	"github.com/free5gc/ausf/internal/sbi/consumer"
-	"github.com/free5gc/ausf/internal/sbi/ueauthentication"
+	"github.com/free5gc/ausf/internal/sbi/processor"
+	"github.com/free5gc/ausf/pkg/app"
 	"github.com/free5gc/ausf/pkg/factory"
-	"github.com/free5gc/util/httpwrapper"
-	logger_util "github.com/free5gc/util/logger"
 )
 
+var AUSF *AusfApp
+
+var _ app.App = &AusfApp{}
+
 type AusfApp struct {
-	cfg     *factory.Config
+	app.App
+	consumer.ConsumerAusf
+	processor.ProcessorAusf
+	sbi.ServerAusf
+
 	ausfCtx *ausf_context.AUSFContext
+	cfg     *factory.Config
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+
+	sbiServer *sbi.Server
+	consumer  *consumer.Consumer
+	processor *processor.Processor
 }
 
-func NewApp(cfg *factory.Config) (*AusfApp, error) {
-	ausf := &AusfApp{cfg: cfg}
+func NewApp(ctx context.Context, cfg *factory.Config, tlsKeyLogPath string) (*AusfApp, error) {
+	ausf := &AusfApp{
+		cfg: cfg,
+		wg:  sync.WaitGroup{},
+	}
 	ausf.SetLogEnable(cfg.GetLogEnable())
 	ausf.SetLogLevel(cfg.GetLogLevel())
 	ausf.SetReportCaller(cfg.GetLogReportCaller())
-
 	ausf_context.Init()
+
+	processor, err_p := processor.NewProcessor(ausf)
+	if err_p != nil {
+		return ausf, err_p
+	}
+	ausf.processor = processor
+
+	consumer, err := consumer.NewConsumer(ausf)
+	if err != nil {
+		return ausf, err
+	}
+	ausf.consumer = consumer
+
+	ausf.ctx, ausf.cancel = context.WithCancel(ctx)
 	ausf.ausfCtx = ausf_context.GetSelf()
+
+	if ausf.sbiServer, err = sbi.NewServer(ausf, tlsKeyLogPath); err != nil {
+		return nil, err
+	}
+	AUSF = ausf
+
 	return ausf, nil
 }
 
-func (a *AusfApp) SetLogEnable(enable bool) {
+func (a *AusfApp) CancelContext() context.Context {
+	return a.ctx
+}
+
+func (a *AusfApp) Consumer() *consumer.Consumer {
+	return a.consumer
+}
+
+func (a *AusfApp) Processor() *processor.Processor {
+	return a.processor
+}
+
+func (a *AusfApp) Context() *ausf_context.AUSFContext {
+	return a.ausfCtx
+}
+
+func (a *AusfApp) Config() *factory.Config {
+	return a.cfg
+}
+
+func (c *AusfApp) SetLogEnable(enable bool) {
```

Comment:
Receiver variable names should be `a`
```suggestion
func (a *AusfApp) SetLogEnable(enable bool) {
```